net/stun/stun.go

```diff
@@ -151,20 +152,21 @@ func foreachAttr(b []byte, fn func(attrType uint16, a []byte) error) error {
 }
 
 // Response generates a binding response.
-func Response(txID TxID, ip net.IP, port uint16) []byte {
-	if ip4 := ip.To4(); ip4 != nil {
-		ip = ip4
-	}
+func Response(txID TxID, addrPort netip.AddrPort) []byte {
+	addr := addrPort.Addr()
+
 	var fam byte
-	switch len(ip) {
-	case net.IPv4len:
+	var bytLen int
+	if addr.Is4() {
 		fam = 1
-	case net.IPv6len:
+		bytLen = 4
+	} else if addr.Is6() {
 		fam = 2
-	default:
+		bytLen = 16
+	} else {
 		return nil
 	}
-	attrsLen := 8 + len(ip)
+	attrsLen := 8 + bytLen
```

Comment:
instead of `bytLen` you could also use `addr.BitLen()/8` (https://go.dev/play/p/_OMhPgqfq8T)

Reply:
I was thinking about that but didn't know if it'd be too messy. I like this better.